src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useCallback } from "react";
+import { AccordionData } from "../AccordionItem/AccordionItem";
+import AccordionItemList from "../AccordionItemList/AccordionItemList";
+import styled from "styled-components";
+
+interface AccordionProps {
+  data: AccordionData[];
+}
+
+const AccordionBlock = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  padding-top: 50px;
+  gap: 10px;
+`;
+const Search = styled.input`
+  width: 500px;
+`;
+
+const Accordion: React.FC<AccordionProps> = ({ data }) => {
+  const [accordionData, setAccordionData] = useState<AccordionData[]>(data);
+  const [searchInput, setSearchInput] = useState<string>("");
+
+  //обновляем состояние каждого элемента
+  const handleToggle = useCallback((id: number) => {
+    setAccordionData((prevData) => {
+      return prevData.map((item) => {
+        if (item.id === id) {
+          return { ...item, open: !item.open };
+        } else if (item.children.length > 0) {
+          return { ...item, children: toggleChildren(item.children, id) };
+        }
+        return item;
+      });
+    });
+  }, []);
+  //обновляем состояние дочерних элементов (children)
+  const toggleChildren = useCallback(
+    (children: AccordionData[], id: number): AccordionData[] => {
+      return children.map((item) => {
+        if (item.id === id) {
+          return { ...item, open: !item.open };
+        } else if (item.children.length > 0) {
```

Comment:
Тут решение аналогичное
```suggestion
        } else if (item.children) {
```